parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/metrics/PCMetrics.java

```diff
@@ -0,0 +1,200 @@
+package io.confluent.parallelconsumer.metrics;
+
+/*-
+ * Copyright (C) 2020-2023 Confluent, Inc.
+ */
+
+import io.micrometer.core.instrument.*;
+import io.micrometer.core.instrument.composite.CompositeMeterRegistry;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.ToDoubleFunction;
+
+import static java.util.Collections.emptyList;
+
+@Slf4j
+/**
+ * Main metrics collection and initialization service.
+ * Singleton - makes it easier to add metrics throughout the code
+ */
+public class PCMetrics {
+    private static PCMetrics instance;
+
+    /**
+     * Meter registry used for metrics - set through init call on singleton initialization. Configurable through
+     * Parallel Consumer Options.
+     */
+    private MeterRegistry meterRegistry;
+
+    /**
+     * Tracking of registered meters for removal from registry on shutdown.
+     */
+    private List<Meter.Id> registeredMeters = new ArrayList<>();
```

Comment:
I am not sure i would want to clear whole registry - as it may be used for other things as well.
For example some logic like:
App managing PC instance (maybe even more than 1) and on whatever condition restarting 1 instance. Metrics registry would be shared across the app itself, any JVM metric collectors etc and PC instances. 
In that case on shut down of PC instance it should only clear the meters that it created and not all meters on registry - at least that is my thinking here.

Reply:
Ah right makes sense.